exist-core/src/main/java/org/exist/xquery/functions/map/MapFunction.java

```diff
@@ -145,9 +145,13 @@ public Sequence eval(final Sequence[] args, final Sequence contextSequence) thro
         return null;
     }
 
-    private Sequence remove(final Sequence[] args) {
+    private Sequence remove(final Sequence[] args) throws XPathException {
         final AbstractMapType map = (AbstractMapType) args[0].itemAt(0);
-        return map.remove((AtomicValue) args[1].itemAt(0));
+        AtomicValue[] keys = new AtomicValue[args[1].getItemCount()];
```

Comment:
Could be replaced with nice lambda/collector expression?

Reply:
@dizzzz Do you have a link to an example?

Reply:
I had a quick look around and I cannot see how this could be applied here.